Rustfmt rayon-core and rayon-demo

ChristopherDavenport opened the pull request:
In response to missing core in the last PR, this applies rustfmt to rayon-core and rayon-demo

stjepang commented on rayon-core/src/configuration.rs:
```diff
@@ -63,7 +63,8 @@ impl Configuration {
     /// the thread's name.
     pub fn set_thread_name<F>(mut self, closure: F) -> Self
-    where F: FnMut(usize) -> String + 'static {
+        where F: FnMut(usize) -> String + 'static
```
Isn't the new style to align `where` with `pub fn`?

ChristopherDavenport replied:
I was running a brand new update of rustfmt so I'm surprised if it didn't come through. Have the new changes made it to release?

nikomatsakis replied:
Yeah, perhaps not...

bluss replied:
It's implemented (rustfmt 8.1) but not the default. where_style = "Rfc" should be it.

Conspicuous config file https://github.com/rust-lang-nursery/rustfmt/blob/master/rfc-rustfmt.toml

stjepang commented on rayon-core/src/future/mod.rs:
```diff
@@ -184,18 +192,18 @@ impl<'scope, F, S> ScopeFuture<'scope, F, S>
+                                                                      result: Ok(Async::NotReady),
+                                                                      canceled: false,
+                                                                  }),
+                                         });
```
This looks pretty bad, IMO :(

nikomatsakis replied:
My policy is generally to run rustfmt unless it produces errors or whatever; but feel free to file a bug in rustfmt if you don't like the output, I guess.

nikomatsakis replied:
(Actually, a better option would to search around for the appropriate conversation in the https://github.com/rust-lang-nursery/fmt-rfcs/ repository...)

nikomatsakis commented:
Can you squash these into one commit (or otherwise remove the merge commit, in any case)?

ChristopherDavenport commented:
Removed Merge Commit and Rebased Off Current Master to resolve merge conflict. 

After rebasing in I reapplied rustfmt on the upstream code. Which has some changes which are in the proper commit.

Sorry for the sloppy history, however it is not present in the final product.

nikomatsakis commented:
@ChristopherDavenport see @nrc's [latest tweet](https://twitter.com/nick_r_cameron/status/846519755599101952): 

> Rustfmt 8.1 released - https://crates.io/crates/rustfmt  Lots of block-indent improvements. Try the emerging RFC-approved style using rfc-rustfmt.toml

Maybe you can add this file into the repo and use those settings?

ChristopherDavenport commented:
So the problem with the new RFC style is that is appears to break macro invocation code by applying trailing commas in places  which breaks compilation.

I removed them in the commit but reapplying cargo fmt seems to reapply these broken fixes. So I'm unsure if we would like to apply such radical fixes which are labeled as very WIP.

nrc commented:
That is https://github.com/rust-lang-nursery/rustfmt/issues/1419 I should fix it within the next week and issue a 0.8.2 release

nikomatsakis commented:
@ChristopherDavenport d'oh

nikomatsakis commented:
maybe we'll wait for the fix then

nrc commented:
I released Rustfmt 0.8.3, it should fix the problem here.

ChristopherDavenport commented:
On it.

ChristopherDavenport commented:
Sorry about all the mess on this PR. I also worked through to get a successful CI test with rustfmt if you would like that to be incorporated as well, meaning moving forward all code would stay current with the standards outlined in rustfmt.toml. I can make that as a separate PR later though.

It would however fix the problem of having to come back and deal with code after the initial commit.

cuviper commented:
In case you missed it, @nikomatsakis was contemplating a bot for ongoing maintenance:
https://internals.rust-lang.org/t/idea-rustfmtbot/5039

ChristopherDavenport commented:
Check, I won't worry about it. You should know that current rustfmt fails to build against 1.12.0. so whatever the solution is will need to insure it doesn't run against that. I'll see if I can try my hands at a bot that does that just for the fun of it. Even if in practice it is done by something else.

ChristopherDavenport commented:
So I have applied rusftmt again to whole project as a single commit. If there is still interest I hope this will be merged soon because rebasing this after every day or so is getting a little tiring.

nikomatsakis commented:
@ChristopherDavenport yeah sorry about that. Thanks! :)

nikomatsakis commented:
@ChristopherDavenport hmm some of the formatting within closures etc looks pretty messed up. I'm inclined to file some rustfmt bugs and maybe revisit this when they've made more progress with the block formatting. I would imagine that [closures are not meant to look like this](https://github.com/nikomatsakis/rayon/pull/280/files#diff-6a14094077d75dad989b9e8f9a440937R103) and [these kinds of if expressions don't seem to line up very well](https://github.com/nikomatsakis/rayon/pull/280/files#diff-2c4fd7e6fd8962b2edf0f53bc2b6dff0R344).

nikomatsakis commented:
Filed two rustfmt bugs. I guess I'll close this for now.

nikomatsakis commented:
Sorry for the confusion @ChristopherDavenport

stjepang commented:
I expressed negative feelings towards rustfmt a while back, but wanna say that it has improved *a lot*. 
These days rustfmt is absolutely fantastic and I use it all the time. Thanks and great job @nrc!
